Model/src/main/java/org/gusdb/wdk/model/user/dataset/event/repo/UserDatasetEventRepo.java

```diff
@@ -0,0 +1,440 @@
+          event.setStatus(optStat.get());
+          unlockEventTable(con);
+          return false;
+        }
+
+        // If we've made it this far, the row is in the "cleanup_ready" status.
+        // Update the status and return true to indicate the row is "locked"
+        event.setStatus(UserDatasetEventStatus.CLEANUP_PROCESSING);
+        updateEventLock(con, event);
+        unlockEventTable(con);
+        return true;
+
+      } catch (SQLException ex) {
+        LOG.warn("Exception caught while attempting to lock event row.  Rolling back transaction.");
+        con.rollback();
+        throw new WdkRuntimeException(ex);
+      }
+    } catch (SQLException ex) {
+      throw new WdkRuntimeException(ex);
+    }
+  }
+
+  /**
+   * Acquires a lock on the events table to prevent race conditions where other
+   * processes attempt to operate on the same row as this process.
+   *
+   * @param con Open JDBC connection.
+   */
+  private void lockEventTable(Connection con) throws SQLException {
+    var sql = "LOCK TABLE " + schema + TABLE_USER_DATASET_EVENT + " IN EXCLUSIVE MODE";
+
+    con.setAutoCommit(false);
+
+    // Use SQLRunner as this query may have to wait for other table locks to
+    // release, making this a potentially "slow" query.
+    new SQLRunner(con, sql, "lock-event-table").executeUpdate();
+  }
+
+  /**
+   * Commits the open JDBC transaction, unlocking the events table for other
+   * processes.
+   *
+   * @param con Open JDBC transaction.
+   */
+  private void unlockEventTable(Connection con) throws SQLException {
+    con.commit();
+  }
+
+  /**
+   * Inserts a new locked event row in the "processing" status.  Additionally
+   * updates the given row instance to be in the "processing" status.
+   *
+   * @param con Open JDBC transaction.
+   * @param row Row to insert.
+   */
+  private void insertEventLock(Connection con, EventRow row) throws SQLException {
+    var sql = "INSERT INTO\n"
+      + "  " + schema + TABLE_USER_DATASET_EVENT + "(\n"
+      + "    event_id\n"
+      + "  , user_dataset_id\n"
+      + "  , handled_time\n"
+      + "  , status\n"
+      + "  , type_name\n"
+      + "  , type_version\n"
+      + "  )"
+      + "VALUES\n"
+      + "  (?, ?, ?, ?, ?, ?)";
+
+    try (var stmt = con.prepareStatement(sql)) {
+      stmt.setLong(1, row.getEventID());
+      stmt.setLong(2, row.getUserDatasetID());
+      stmt.setObject(3, row.getHandledTime(), Types.TIMESTAMP);
+      stmt.setString(4, row.getStatus().internalValue());
+      stmt.setString(5, row.getType().getName());
+      stmt.setString(6, row.getType().getVersion());
+
+      stmt.executeUpdate();
+    }
+  }
+
+  /**
+   * Updates the given row's status in the DB (and the row instance) to
+   * "processing".
+   *
+   * @param con Open JDBC transaction.
+   * @param row Row to update.
+   */
+  private void updateEventLock(Connection con, EventRow row) throws SQLException {
```

Comment:
more on the theme of "event locks" being confusing...  this method really is just updateEventStatus().  i think overall the code/comments should make clear that there is no event lock;  we're using the status to claim the event and then release our claim

Reply:
Updated